pkg/controller/generic/transform/controller.go

```diff
@@ -235,6 +256,21 @@ func (ctrl *Controller[Input, Output]) Run(ctx context.Context, r controller.Run
 			return err
 		}
 
+		if state.requeue && ctrl.options.requeueInterval != 0 {
```

Comment:
we could in theory build a proper requeue channel with all bells and whistles, for a specific ID if you'd like to?